publisher-sdk/src/main/java/com/criteo/publisher/logging/LogMessage.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ *    Copyright 2020 Criteo
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.criteo.publisher.logging
+
+import android.util.Log
+
+internal data class LogMessage(
+    val level: Int = Log.INFO,
+    private val message: () -> String?,
+    val throwable: Throwable? = null
+) {
+  val formattedMessage: String? by lazy {
+    message()
+  }
+
+  override fun equals(other: Any?): Boolean {
```

Comment:
Any reasons why there are equals/hashcode here? The kotlin compiler already generates them for data classes:

```
  public int hashCode() {
      int var10000 = this.level * 31;
      Function0 var10001 = this.message;
      var10000 = (var10000 + (var10001 != null ? var10001.hashCode() : 0)) * 31;
      Throwable var1 = this.throwable;
      return var10000 + (var1 != null ? var1.hashCode() : 0);
   }

   public boolean equals(@Nullable Object var1) {
      if (this != var1) {
         if (var1 instanceof LogMessage) {
            LogMessage var2 = (LogMessage)var1;
            if (this.level == var2.level && Intrinsics.areEqual(this.message, var2.message) && Intrinsics.areEqual(this.throwable, var2.throwable)) {
               return true;
            }
         }

         return false;
      } else {
         return true;
      }
   }
```

Reply:
I want the equal/hashCode/toString to be on the `formattedMessage`.